lib/matplotlib/figure.py

```diff
@@ -1838,21 +1838,27 @@ def subplot_mosaic(self, mosaic, *, sharex=False, sharey=False,
             Defines the relative widths of the columns. Each column gets a
             relative width of ``width_ratios[i] / sum(width_ratios)``.
             If not given, all columns will have the same width.  Equivalent
-            to ``gridspec_kw={'width_ratios': [...]}``.
+            to ``gridspec_kw={'width_ratios': [...]}``. This argument applies
+            only to the outer layout and will not be passed to the inner
+            layout.
 
         height_ratios : array-like of length *nrows*, optional
             Defines the relative heights of the rows. Each row gets a
             relative height of ``height_ratios[i] / sum(height_ratios)``.
             If not given, all rows will have the same height. Equivalent
-            to ``gridspec_kw={'height_ratios': [...]}``.
+            to ``gridspec_kw={'height_ratios': [...]}``. This argument
+            applies only to the outer layout and will not be passed to the
+            inner layout.
 
         subplot_kw : dict, optional
             Dictionary with keywords passed to the `.Figure.add_subplot` call
             used to create each subplot.
 
         gridspec_kw : dict, optional
             Dictionary with keywords passed to the `.GridSpec` constructor used
-            to create the grid the subplots are placed on.
+            to create the grid the subplots are placed on. The 'width_ratios'
+            and 'height_ratios' keys, if present, will be removed from this
+            dictionary before it is passed to any inner layouts.
```

Comment:
I guess this is a discussion point, but I don't think any of the other gridspec keywords (`left`, `top` etc) make much sense to pass down a level either.   Wouldn't it just be easier to say "these keyword arguments are only used for the top-level gridspec for nested mosaic layouts; more complicated layouts should use `.subfigure` and `.subplots` directly."

Reply:
Yeah I'd agree with that. I guess at present there's no way of specifying those args with any kind of granularity. I'd argue that it _might_ still be useful to be able to set `wspace` and `hspace` for the entire mosaic, but I'd agree that anything more complicated should probably be constructed directly, and keeping just those args might complicate the docstring even further.

If we want to pass this dict only to the top-level, I can completely remove this behaviour and there would be no need for the new dict. If we want to keep `wspace` and `hspace` I guess the new dict could be adjusted to be a whitelist of args, rather than a blacklist (as it is currently). Which do you think would be better?

Reply:
I think wspace and hspace are esoteric enough to ask folks to do manually.  Otoh we should get a couple more opinions before doing anything drastic.

Reply:
Yeah, fair enough.

I'll leave this as-is for now to give others a chance to chime-in.

Reply:
I stongly favor not passing anything of `gridspec_kw` down to inner layouts. This is conceptually simpler. And IMHO it's more common to want different (or default) parameters for the inner layouts. These use cases would not be addressed (or prevented) by the propagation.

Reply:
Okay, I'll modify my PR ASAP